### PR TITLE
ESQL: Add README.md to docs

### PR DESCRIPTION
**Pull request by nik9000**
This explains how to run the tests that build the docs. I tried to add it in #106577 but the sync code deleted it. So I fixed that too.

### CODE REVIEW

**Review by mark-vieira — APPROVED** [review]

**Comment by nik9000** [comment]
Thanks @mark-vieira !

File: docs/reference/esql/functions/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+The files in these subdirectories and generated by ESQL's test suite:
```
Comment (leemthompo) [inline docs/reference/esql/functions/README.md:1]:
```suggestion
The files in these subdirectories are generated by ESQL's test suite:
```

**Review by leemthompo — APPROVED** [review]
Thanks @nik9000!

File: docs/reference/esql/functions/README.md (hunk 1 above)
Comment (nik9000) [reply docs/reference/esql/functions/README.md:1]: I hadn't seen this before the merge landed. Here's the fix: https://github.com/elastic/elasticsearch/commit/d6d1edd52959f358ce1df438d01ddf03ef8b5c21

**Pull request merged**